Automattic/wp-calypso
# Checkout: Add Google Pay payment method

#### Changes proposed in this Pull Request

This will add a new payment method for Google Pay to checkout (when available).

<img width="565" alt="Screen Shot 2021-05-06 at 7 29 43 PM" src="https://user-images.githubusercontent.com/2036909/117377581-96543200-aea1-11eb-8920-ccca06d2323a.png">

<img width="596" alt="Screen Shot 2021-04-28 at 6 20 42 PM" src="https://user-images.githubusercontent.com/2036909/116479974-85c30c80-a84e-11eb-8aa1-f61ec5bdfcfc.png">

Fixes 154-gh-Automattic/payments-shilling 

Currently, this is behind a feature flag which is only enabled on non-production environments.

#### Testing instructions

You can test this using the calypso.live branch (see comments below) without any additional work, but if you are running calypso on localhost, follow these steps first:

- Install [localtunnel](https://github.com/localtunnel/localtunnel) on your computer, probably with `npm install -g localtunnel`
- Apply D60830-code to allow using localtunnel and sandbox the API.
- Make sure calypso is running on calypso.localhost:3000
- Run `lt -p 3000 -h https://a8c-localtunnel.cyou` in the wp-calypso repository.
- You should get an https URL printed to the console. Open that URL in Google Chrome and use it for the below testing steps.

To test the payment method, you'll need to be using Google Chrome and already have a wallet with at least one active card stored in Google Pay. If you meet that criteria, you may proceed with the following instructions. Otherwise you will not see "Google Pay" as an option.

If your API is sandboxed, you can use a real credit card and will not be charged for the transaction because Stripe is operating in test mode.

- Add a product to your cart and visit checkout.
- Verify that "Google Pay" is one of the options in the list.
- Click "Google Pay", and then click the submit button at the bottom of checkout to confirm the payment.
- Verify that you see the Google Pay sheet appear in your browser and that the price there matches the price in checkout.
- Click to confirm the purchase.
- Verify that the purchase is successful and that you are redirected away from checkout to a thank-you page or with a thank-you notification as normal.
